test/framework/cluster.go

```diff
@@ -1673,22 +1656,13 @@ func (e *ClusterE2ETest) CombinedAutoScalerMetricServerTest(autoscalerName strin
 func (e *ClusterE2ETest) ValidateClusterState() {
 	e.T.Logf("Validating cluster %s", e.ClusterName)
 	ctx := context.Background()
-	err := retrier.Retry(60, 5*time.Second, func() error {
-		return e.buildClusterValidator(ctx)
-	})
-	if err != nil {
-		e.T.Fatalf("failed to build cluster validator %v", err)
-	}
-
-	if e.ClusterConfig.Cluster.IsManaged() {
-		e.clusterValidator.WithWorkloadClusterValidations()
-	}
-	e.clusterValidator.WithExpectedObjectsExist()
-
-	providerValidations := e.Provider.ClusterValidations()
-	e.clusterValidator.WithValidations(providerValidations...)
-
-	if err := e.clusterValidator.Validate(ctx); err != nil {
+	e.buildClusterStateValidationConfig(ctx)
+	clusterStateValidator := newClusterStateValidator(e.clusterStateValidationConfig)
+	validations := validationsForExpectedObjects()
+	providerValidations := e.Provider.ClusterStateValidations()
+	validations = append(validations, providerValidations...)
+	clusterStateValidator.WithValidations(validations...)
```

Comment:
nit, but I think it's a bit cleaner
```suggestion
	clusterStateValidator := newClusterStateValidator(e.clusterStateValidationConfig)
	clusterStateValidator.WithValidations(validationsForExpectedObjects()...)
	clusterStateValidator.WithValidations(e.Provider.ClusterStateValidations()...)
```